### PR TITLE
Output the outstanding states during state scan

### PR DESCRIPTION
**Pull request by carise**

### CODE REVIEW

File: src/main/java/com/google/cloud/security/scanner/common/CloudUtil.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+package com.google.cloud.security.scanner.common;
+
+import com.google.appengine.api.utils.SystemProperty;
+
+public class CloudUtil {
+  
+  public static boolean willExecuteOnCloud() {
+    return SystemProperty.environment.value() == SystemProperty.Environment.Value.Production;
+  }
+}
```
Comment (blueandgold) [inline src/main/java/com/google/cloud/security/scanner/common/CloudUtil.java]: nit: newline

File: src/main/java/com/google/cloud/security/scanner/pipelines/LiveStateChecker.java [hunk 2]
```diff
@@ -115,18 +141,58 @@ LiveStateChecker appendAssertContains(String[] expectedOutputs) {
...
             .of(new TagStateWithSource(StateSource.LIVE)));
 
-    // Join the two known-good and the live halves.
     PCollectionView<Map<GCPResource, KV<StateSource, GCPResourceState>>> knownGoodStatesView =
         taggedKnownGoodStates.apply(View.<GCPResource, KV<StateSource, GCPResourceState>>asMap());
+    PCollectionView<Map<GCPResource, KV<StateSource, GCPResourceState>>> liveStatesView =
+        taggedLiveStates.apply(View.<GCPResource, KV<StateSource, GCPResourceState>>asMap());
+
+    // Find outstanding states (known good with no matching live/live with no matching known-good)
+    PCollection<KV<String, GCPResource>> outstandingKnownGoodStates =
```
Comment (blueandgold) [inline src/main/java/com/google/cloud/security/scanner/pipelines/LiveStateChecker.java]: "outstanding" is a bit ambiguous; can we try to find a better naming?
Maybe something along the line of:
`unmatchedKnownGoodStates`
`unmatchedLiveStates`
`FindUnmatchedStates()`
`mergedUnmatchedStates`

**Comment by blueandgold** [comment]
Looks great!!  And thanks for adding the test!  :100:

**Comment by blueandgold** [comment]
👍

**Comment by carise** [comment]
@blueandgold PTAL, I had to change the pipeline to be static because of a serialization issue with my DoFn (similar issue: http://stackoverflow.com/questions/39842598/unable-to-run-a-job-with-outputtags). This is for my recent commit, 25da60f

**Comment by blueandgold** [comment]
Nice!  This is subtle, yet makes sense when understanding what's happening.  💯

**Pull request merged**